pkg/crypto/tinkcrypto/primitive/jwe/ecdhes/ecdh_aead_public_key_manager.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package ecdhes
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/tink/go/core/registry"
+	"github.com/google/tink/go/keyset"
+	"github.com/google/tink/go/subtle/hybrid"
+	tinkpb "github.com/google/tink/proto/tink_go_proto"
+
+	sbutleecdhes "github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto/primitive/jwe/subtle/ecdhes"
+	ecdhespb "github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto/primitive/proto/ecdhes_aead_go_proto"
+)
+
+const (
+	ecdhesPublicKeyVersion = 0
+	ecdhesPublicKeyTypeURL = "type.googleapis.com/google.crypto.tink.EcdhesAeadPublicKey"
+)
+
+// common errors
+var errInvalidECDHESPublicKey = fmt.Errorf("ecdhes_public_key_manager: invalid key")
+
+// ecdhesPublicKeyManager is an implementation of KeyManager interface.
+// It generates new ECDHESPublicKey keys and produces new instances of ECDHESPublicKey subtle.
+type ecdhesPublicKeyManager struct{}
+
+// Assert that ecdhesPublicKeyManager implements the KeyManager interface.
+var _ registry.KeyManager = (*ecdhesPublicKeyManager)(nil)
+
+// newECDHESPublicKeyManager creates a new aesGcmKeyManager.
+func newECDHESPublicKeyManager() *ecdhesPublicKeyManager {
+	return new(ecdhesPublicKeyManager)
+}
+
+// Primitive creates an ECDHESPublicKey subtle for the given serialized ECDHESPublicKey proto.
+func (km *ecdhesPublicKeyManager) Primitive(serializedKey []byte) (interface{}, error) {
+	if len(serializedKey) == 0 {
+		return nil, errInvalidECDHESPublicKey
+	}
+
+	key := new(ecdhespb.EcdhesAeadPublicKey)
+
+	err := proto.Unmarshal(serializedKey, key)
+	if err != nil {
+		return nil, errInvalidECDHESPublicKey
+	}
+
+	err = km.validateKey(key)
+	if err != nil {
+		return nil, errInvalidECDHESPublicKey
+	}
+
+	curve, err := hybrid.GetCurve(key.Params.KemParams.CurveType.String())
+	if err != nil {
+		return nil, err
+	}
+
+	pub := hybrid.ECPublicKey{
+		Curve: curve,
+		Point: hybrid.ECPoint{
+			X: new(big.Int).SetBytes(key.X),
+			Y: new(big.Int).SetBytes(key.Y),
+		},
+	}
+
+	rDem, err := newRegisterECDHESEncHelper(key.Params.DemParams.AeadDem)
+	if err != nil {
+		return nil, err
+	}
+
+	ptFormat := key.Params.EcPointFormat.String()
+
+	// TODO add support for multiple 'recipient' keys somehow.
+	// 1 key template is for 1 recipient only. A possible solution would be to update Encrypt() interface to take
```

Comment:
Updating tink interfaces doesn't seem very generic.

Reply:
The interface is created in this change (see `pkg/crypto/tinkcrypto/primitive/jwe/ecdhes_decrypt.go` below), each primitive has its own interface.

There are two entries to Tink when getting a new primitive and executing it, one is when creating the key template as follows:
`kh, err := keyset.NewHandle(AES256GCMECDHESKeyTemplate())`

One option would be to add recipients argument to `AES256GCMECDHESKeyTemplate()`

The other option is when executing the primitive operation, the `Encrypt()` function interface as mentioned in the code comment.

But that would be for another issue, hence the TODO, I'll create an issue for this.

Reply:
Isn't the `Encrypt` method a defined interface? (I assume unlike the constructor `AES256GCMECDHESKeyTemplate`)

Reply:
BTW if you put recipients into the constructor, why would we still need JWE return values.

Reply:
JWE reference is now removed from this change and now supports multiple recipients exchange in the crypto side.